checks/sanity_check.go

```diff
@@ -24,12 +24,20 @@ var (
 // sanityCheck checks for configuration that is either misplaced or in conflict
 func sanityCheck(ctx context.Context, conf *config.Configuration, res *api.RegistrationResponse, _ runtimeConfig) error {
 	if util.AnyEnvVarsExist(awsLogIngestionEnvVars) {
-		return fmt.Errorf("Environment varaible '%s' is used by aws-log-ingestion and has no effect here. Recommend unsetting this environment variable within this function.", util.AnyEnvVarsExistString(awsLogIngestionEnvVars))
+		return fmt.Errorf("Environment variable '%s' is used by aws-log-ingestion and has no effect here. Recommend unsetting this environment variable within this function.", util.AnyEnvVarsExistString(awsLogIngestionEnvVars))
 	}
 
 	if credentials.IsSecretConfigured(ctx, conf) && util.EnvVarExists("NEW_RELIC_LICENSE_KEY") {
 		return fmt.Errorf("There is both a AWS Secrets Manager secret and a NEW_RELIC_LICENSE_KEY environment variable set. Recommend removing the NEW_RELIC_LICENSE_KEY environment variable and using the AWS Secrets Manager secret.")
 	}
 
+	if credentials.IsParameterConfigured(ctx, conf) && util.EnvVarExists("NEW_RELIC_LICENSE_KEY") {
+		return fmt.Errorf("There is both a AWS Parameter Store parameter and a NEW_RELIC_LICENSE_KEY environment variable set. Recommend removing the NEW_RELIC_LICENSE_KEY environment variable and using the AWS Parameter Store parameter.")
+	}
+
+	if credentials.IsSecretConfigured(ctx, conf) && credentials.IsParameterConfigured(ctx, conf) {
+		return fmt.Errorf("There is both a AWS Secrets Manager secret and a AWS Parameter Store parameter set. Recommend using just one.")
+	}
```

Comment:
I think this is a growing performance cost, as every call to `credentials.IsXConfigured()` makes an API call. A few suggestions:
1. We might as well store the secret and param results in a variable, to avoid making 2x the API calls.
2. Inside the `credentials.IsXConfigured()` methods, we might as well return early if the name/ID is empty, as we know the request will return an error.